### PR TITLE
feat(embodied): update metaworld environment and clean codes for openpi

### PR DESCRIPTION
**Pull request by chenkang455**
This PR is designed to update some code for Metaworld and clean some code for openpi for better readability.
### TODO
- [ ] Fix variables with the same name but different values ​​that exist in `OpenPi0Config` and `actor.model`, such as action_dim. Now we use `actor.model.openpi` for isolation in the YAML.

### Description

1. I update the pi05 config for metaworld MT50.
2. I update the metaworld environment to support ML45 Task.
3. I cleaned some code for openpi.
4. Update a Python code for converting .pt weights to .modeltensor weights.

### Motivation and Context

 
1. Metaworld results are updated.
2. The current openpi code has a lot of redundant code during the debug stage, I cleaned them for better readability

### How has this been tested?

Check the doc here: https://rlinf.readthedocs.io/en/latest/rst_source/examples/metaworld.html

### Additional information (optional, e.g., figures and logs):
image
Results are evaluated on the 450-step checkpoint. The training curve below is still increasing.

image

I modify the initial SFT checkpoint of the original one and place them in `/mnt/mnt/public/chenkang/openpi-main/checkpoints/sft/pi0_metaworld/metaworld_32_10k/9999` (pi0) and `/mnt/mnt/public/chenkang/openpi-main/checkpoints/sft/pi05_metaworld/metaworld_32_10k/9999` (pi05). Can you help me update the pi0 model and upload the pi05 model? @andylin-hao Thanks a lot!

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Documentation update (Document-only update)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: toolkits/ckpt_convertor/convert_state_dict_to_safetensor.py
Comment (chenkang455) [inline toolkits/ckpt_convertor/convert_state_dict_to_safetensor.py:1]: This code is used for converting the .pt model to safetensor format for openpi models.

File: rlinf/workers/actor/fsdp_actor_worker.py [hunk 1]
```diff
@@ -834,9 +834,6 @@ def run_training(self):
                             use_cache=False,
                         )
 
-                    if self.cfg.actor.model.model_name in ["openpi"]:
```
Comment (chenkang455) [inline rlinf/workers/actor/fsdp_actor_worker.py]: This part is done in the rollout stage so that the interface is aligned with other methods.

File: rlinf/workers/actor/fsdp_actor_worker.py [hunk 2]
```diff
@@ -864,7 +861,7 @@ def run_training(self):
                     entropy_loss = torch.tensor(0.0, device=torch.cuda.current_device())
                     if (
                         self.cfg.algorithm.entropy_bonus > 0
-                        and not kwargs.critic_warmup
+                        and not kwargs["critic_warmup"]
```
Comment (chenkang455) [inline rlinf/workers/actor/fsdp_actor_worker.py:867]: When setting the entorpy_bonus, i discover that this will cause a bug, kwargs is a dict.

File: rlinf/models/embodiment/openpi_action_model.py
Comment (chenkang455) [inline rlinf/models/embodiment/openpi_action_model.py:1]: This part mainly cleans the openpi code. @LiuZhihao2022 can help review it?

File: examples/embodiment/config/metaworld_50_ppo_openpi_pi05.yaml [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    total_training_steps: ${runner.max_epochs}
+    lr_warmup_steps: -1
+    warmup_style: cosine
+    min_lr_ratio: 0.1
+    num_cycles: 0.5
+
+    
+    
+
+
```
Comment (QuanluZhang) [inline examples/embodiment/config/metaworld_50_ppo_openpi_pi05.yaml]: too many blank lines.

File: rlinf/models/embodiment/openpi_action_model.py [hunk 4]
```diff
@@ -32,33 +33,39 @@
 @dataclass(frozen=True)
 class OpenPi0Config(Pi0Config):
     # config for rl
-    noise_level: bool = 0.5
-    action_chunk: int = 5
-    train_expert_only: bool = False
-    action_env_dim: int = 7  # for libero
-    num_steps: int = 10
+    simulator_type: str = "libero"  # libero, metaworld, maniskill
     noise_method: str = "flow_sde"  # flow_sde, flow_cps
```
Comment (QuanluZhang) [inline rlinf/models/embodiment/openpi_action_model.py]: the comment is not correct, right? "flow_cps" should be "flow-noise"?

File: rlinf/models/embodiment/openpi_action_model.py [hunk 5]
```diff
@@ -81,19 +88,28 @@ def __init__(
...
         self.sample_actions = sample_actions_func
+        self.global_step = 0
+        # assert
+        assert not (self.config.double_layer and self.config.joint_logprob), (
+            "double_layer and joint_logprob can not be set at the same time"
+        )
+
+        # camera number
+        if self.config.simulator_type == "metaworld":
+            self.camera_num = 1
```
Comment (XuS1994) [inline rlinf/models/embodiment/openpi_action_model.py]: `camera_num` is a yaml config in earlier version. Has been deleted?

File: rlinf/models/embodiment/openpi_action_model.py [hunk 6]
```diff
@@ -81,19 +88,28 @@ def __init__(
...
+            raise ValueError(
+                f"Not supported simulator type: {self.config.simulator_type} for openpi action model"
+            )
 
         # rl model init
         if self.config.value_after_vlm:
-            assert not self.config.joint_logprob, (
-                "joint logprob is not tested for value_after_vlm mode"
-            )
-            assert not (self.config.double_layer and self.config.joint_logprob), (
```
Comment (XuS1994) [inline rlinf/models/embodiment/openpi_action_model.py:90]: Why delete assert part?

File: toolkits/ckpt_convertor/convert_state_dict_to_safetensor.py
Comment (qurakchin) [reply toolkits/ckpt_convertor/convert_state_dict_to_safetensor.py:1]: which model will use .pt state dict? most huggingface models are already using safetensor now.

File: rlinf/models/embodiment/openpi_action_model.py [hunk 7]
```diff
@@ -368,21 +378,11 @@ def sample_actions(
...
-                    else:
-                        denoise_inds = torch.tensor(
-                            [random.randint(0, num_steps - 1)] * num_steps
-                        )
-                elif self.config.noise_method == "flow_cps":
-                    # the last denoising step of the flow-cps is deterministic
+                if self.config.ignore_last:
                     denoise_inds = torch.tensor(
-                        [random.randint(0, num_steps - 1)] * num_steps
+                        [random.randint(0, num_steps - 2)] * num_steps
```
Comment (XuS1994) [inline rlinf/models/embodiment/openpi_action_model.py:378]: Why use -2 replace -1

**Review by XuS1994 — APPROVED** [review]

File: rlinf/models/embodiment/openpi_action_model.py (hunk 5 above)
Comment (chenkang455) [reply rlinf/models/embodiment/openpi_action_model.py]: This camera number is a new added parameter designed to mask image tokens.
image

File: rlinf/models/embodiment/openpi_action_model.py (hunk 6 above)
Comment (chenkang455) [reply rlinf/models/embodiment/openpi_action_model.py:90]: This part has been moved to the front and the joint-logprob has been tested on the value_after_vlm mode.
image

File: rlinf/models/embodiment/openpi_action_model.py (hunk 7 above)
Comment (chenkang455) [reply rlinf/models/embodiment/openpi_action_model.py:378]: The modified version looks like below, it is controlled by ignore_last param.

image

File: rlinf/models/embodiment/openpi_action_model.py (hunk 4 above)
Comment (chenkang455) [reply rlinf/models/embodiment/openpi_action_model.py]: get it

File: toolkits/ckpt_convertor/convert_state_dict_to_safetensor.py
Comment (chenkang455) [reply toolkits/ckpt_convertor/convert_state_dict_to_safetensor.py:1]: I didn't find a script to convert the dcp to safetensor. Convert_dcp_to_state_dict.py converts the dcp to .pt, so i add this file to convert the .pt to safetensor.

File: rlinf/models/embodiment/openpi_action_model.py [hunk 8]
```diff
@@ -32,33 +33,39 @@
 @dataclass(frozen=True)
 class OpenPi0Config(Pi0Config):
     # config for rl
-    noise_level: bool = 0.5
-    action_chunk: int = 5
-    train_expert_only: bool = False
-    action_env_dim: int = 7  # for libero
-    num_steps: int = 10
+    simulator_type: str = "libero"  # libero, metaworld, maniskill
```
Comment (guozhen1997) [inline rlinf/models/embodiment/openpi_action_model.py]: It appears that `simulator_type` is only used to set the `camera_num`. It's recommended to remove this variable from the model configuration and replace it with either `num_images_in_inputs` or `camera_num`.

File: rlinf/models/embodiment/openpi_action_model.py (hunk 4 above)
Comment (guozhen1997) [inline rlinf/models/embodiment/openpi_action_model.py]: Suggest changing `noise_method` to an `enum` variable instead of a `str`.

File: rlinf/models/embodiment/openpi_action_model.py (hunk 8 above)
Comment (chenkang455) [reply rlinf/models/embodiment/openpi_action_model.py]: image
simulator_type will also affect the input camera names like here

File: rlinf/models/embodiment/openpi_action_model.py (hunk 4 above)
Comment (chenkang455) [reply rlinf/models/embodiment/openpi_action_model.py]: It's a good suggestion, pi and gr00t models have the same noise methods, I suggest that we reformulate this name in the further and manage them together?

**Review by XuS1994 — APPROVED** [review]

File: rlinf/models/embodiment/openpi_action_model.py (hunk 8 above)
Comment (guozhen1997) [reply rlinf/models/embodiment/openpi_action_model.py]: You can use `"wrist_images" in obs` to replace `simulator_type=="libero"`, which is more general and can support other simulators. My key point is that the parameters in the model configuration should be simple and precise.

File: rlinf/models/embodiment/openpi_action_model.py (hunk 4 above)
Comment (guozhen1997) [reply rlinf/models/embodiment/openpi_action_model.py]: ok

File: rlinf/models/embodiment/openpi_action_model.py (hunk 8 above)
Comment (chenkang455) [reply rlinf/models/embodiment/openpi_action_model.py]: Get it, but i have one another interface in the `get_model` function, where I have to select the model config, is there a better way to manage it?
image

File: docs/source-zh/rst_source/examples/metaworld.rst [hunk 9]
```diff
@@ -77,10 +77,12 @@
    # 方法 1: 使用 git clone
    git lfs install
    git clone https://huggingface.co/RLinf/RLinf-Pi0-MetaWorld
+   git clone https://huggingface.co/RLinf/RLinf-Pi05-MetaWorld
 
    # 方法 2: 使用 huggingface-hub
    pip install huggingface-hub
    hf download RLinf/RLinf-Pi0-MetaWorld
+   hf download RLinf/RLinf-Pi05-MetaWorld
```
Comment (andylin-hao) [inline docs/source-zh/rst_source/examples/metaworld.rst:85]: @chenkang455 Where can I find the model? I can help upload it to huggingface

Comment (chenkang455) [reply docs/source-zh/rst_source/examples/metaworld.rst:85]: @andylin-hao I modify the initial SFT checkpoint of the original one and place them in /mnt/mnt/public/chenkang/openpi-main/checkpoints/sft/pi0_metaworld/metaworld_32_10k/9999 (pi0) and /mnt/mnt/public/chenkang/openpi-main/checkpoints/sft/pi05_metaworld/metaworld_32_10k/9999 (pi05). Can you help me update the pi0 model and upload the pi05 model? @andylin-hao Thanks a lot!

Comment (andylin-hao) [reply docs/source-zh/rst_source/examples/metaworld.rst:85]: @chenkang455 Done!
